### PR TITLE
feat/qguiapplication

### PR DESCRIPTION
**Pull request by pinebit**

### CODE REVIEW

**Review by richard-ramos — APPROVED** [review]

**Review by jrainville — APPROVED** [review]

File: lib/src/DOtherSide.cpp [hunk 1]
```diff
@@ -130,8 +130,7 @@ void dos_qapplication_initialize_opengl()
 void dos_qguiapplication_create()
 {
     static int argc = 1;
-    static char empty[1] = {0};
-    static char *argv[] = {empty};
+    static char *argv[] = {"Status"};
```
Comment (emizzle) [inline lib/src/DOtherSide.cpp:133]: Should this be `"Status Desktop"`?

File: lib/src/DOtherSide.cpp [hunk 2]
```diff
@@ -158,10 +157,21 @@ void dos_qguiapplication_quit()
...
+{
+    DockClicker *dockClicker = new DockClicker(vptr);
+    qGuiApp->installEventFilter(dockClicker);
+}
+
+void dos_qapplication_create()
 {
     static int argc = 1;
-    static char *argv[] = {appName};
+    static char *argv[] = {"Status"};
```
Comment (emizzle) [inline lib/src/DOtherSide.cpp:174]: Should we maybe create a constant so it can be used in both places?

**Review by emizzle — COMMENTED** [review]
Nice work, just a couple small comments.

File: lib/src/DOtherSide.cpp (hunk 1 above)
Comment (emizzle) [reply lib/src/DOtherSide.cpp:133]: OK, I see from [the status-desktop PR comment](https://github.com/status-im/status-desktop/pull/2953/files#r672265160) that this is meant to represent the executable name. I believe we have two different names, "StatusDev" and "Status", depending on the build env, but I also could have the wrong idea about what this string is meant to represent.

Comment (pinebit) [reply lib/src/DOtherSide.cpp:133]: Not a name, but full path of the executable.
In our case we just pass any non empty string to not make Qt confused due to an empty argv[0]
In fact, this is not used by Qt, except the case when you did not set applicationName property explicitly, in this case Qt will be using argv[0] as the application name. But we always set that property in mail.qml, so this argv[0] will always be unused.

Comment (emizzle) [reply lib/src/DOtherSide.cpp:133]: Ah ok, all makes sense now. Thank you! 👍

**Pull request merged**